nncf/torch/prepare_for_inference.py

```diff
@@ -0,0 +1,213 @@
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import copy
+
+import numpy as np
+import torch
+from torch.quantization.fake_quantize import FakeQuantize
+
+from nncf.api.compression import CompressionAlgorithmController
+from nncf.torch.composite_compression import CompositeCompressionAlgorithmController
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.pruning.filter_pruning.layers import FilterPruningMask
+from nncf.torch.pruning.operations import PT_PRUNING_OPERATOR_METATYPES
+from nncf.torch.pruning.operations import ModelPruner
+from nncf.torch.pruning.operations import PrunType
+from nncf.torch.quantization.layers import AsymmetricQuantizer
+from nncf.torch.quantization.layers import BaseQuantizer
+from nncf.torch.quantization.layers import SymmetricQuantizer
+
+SUPPORTED_ALGORITHMS = ["quantization", "filter_pruning"]
+
+
+def prepare_for_inference(
+    compressed_model: NNCFNetwork, compressed_ctrl: CompressionAlgorithmController, save_original_model: bool = False
+) -> NNCFNetwork:
+    """
+    Prepare NNCFNetwork for inference
+      - for quantization algorithm replace Replace NNCF quantizers modules to FakeQuantize.
+      - for pruning_filter algorithm prune module by filling weights and bias with zeros.
+
+    :param compressed_model: Compressed model.
+    :param compressed_ctrl: Compression controller.
+    :param save_original_model: `True` means that a copy of the model will be modified.
+
+    :return NNCFNetwork: Converted model.
+    """
+
+    if not isinstance(compressed_model, NNCFNetwork):
+        raise ValueError(f"Expected type of compressed_model is NNCFNetwork, but got {type(compressed_model)}.")
+
+    if save_original_model:
+        compressed_model = copy.deepcopy(compressed_model)
+
+    compressed_model.train(False)
+
+    if isinstance(compressed_ctrl, CompositeCompressionAlgorithmController):
+        ctrls = compressed_ctrl.child_ctrls
+    else:
+        ctrls = [compressed_ctrl]
+
+    # Check supported algorithms
+    for controller in ctrls:
+        if controller.name not in SUPPORTED_ALGORITHMS:
+            raise RuntimeError(f"Function prepare_for_inference supports only {SUPPORTED_ALGORITHMS} algorithms.")
+
+    # Strip the model
+    for controller in ctrls:
+        compressed_model = controller.strip_model(compressed_model)
+
+    # Prepare the model for inference
+    for controller in ctrls:
+        if controller.name == "quantization":
+            replace_quantizer_to_native_module(compressed_model)
+        if controller.name == "filter_pruning":
+            graph = compressed_model.get_original_graph()
+            ModelPruner(compressed_model, graph, PT_PRUNING_OPERATOR_METATYPES, PrunType.FILL_ZEROS).prune_model()
+
+    clean_operators(compressed_model)
+
+    return compressed_model
+
+
+def clean_operators(model: NNCFNetwork) -> None:
```

Comment:
we need to separate what is intend to be used as external API from internal helpers.